banyand/liaison/grpc/property.go

```diff
@@ -405,11 +411,73 @@ func (ps *propertyServer) Query(ctx context.Context, req *propertyv1.QueryReques
 	return &propertyv1.QueryResponse{Properties: properties, Trace: trace}, nil
 }
 
+func (ps *propertyServer) repairPropertyIfNeed(ctx context.Context, entity string, p *propertyWithCount, groups map[string]*commonv1.Group) error {
+	// make sure have the enough replicas
+	group := groups[p.Metadata.Group]
+	if group == nil {
+		return errors.Errorf("group %s not found", p.Metadata.Group)
+	}
+	copies, ok := ps.groupRepo.copies(p.Metadata.GetGroup())
+	if !ok {
+		return errors.New("failed to get group copies")
+	}
+	if copies == uint32(len(p.existNodes)) {
+		return nil
+	}
+	id, err := partition.ShardID(convert.StringToBytes(entity), group.ResourceOpts.ShardNum)
+	if err != nil {
+		return err
+	}
+	// building the repair data
+	repairReq := &propertyv1.InternalRepairRequest{ShardId: uint64(id)}
+	if p.deletedTime > 0 { // building the delete request
+		repairReq.Operation = &propertyv1.InternalRepairRequest_Delete{
```

Comment:
If the deletion's version is less than the latest version on the data node, the deletion should be dropped. I recommend using a unique repair request that contains a "deleted" field.

Reply:
The property in the liaison node will only use the latest version: https://github.com/apache/skywalking-banyandb/pull/686/files#diff-ec9aafdd27b3b36de2fae1ed134371dd730232bbd344d77197eebd1b0927221eR375
So, if the deletion's version is less than the latest version, it will still be an applied request, not a deleted request. When the applied request is sent to the data node, the data will be deleted and all older versions, and apply the new one.

Reply:
Other users may trigger additional writing during this process, making it impossible to guarantee that the data remains consistent after the query.

It's important to remember that all data can change over time, which is why distributed operations can be challenging.